Kernel/Thread.h

```diff
@@ -263,16 +263,20 @@ class Thread {
 
     bool is_stopped() const { return m_state == Stopped; }
     bool is_blocked() const { return m_state == Blocked; }
-    bool in_kernel() const { return (m_tss.cs & 0x03) == 0; }
+    bool in_kernel() const { return (get_register_dump_from_stack().cs & 0x03) == 0; }
 
-    u32 frame_ptr() const { return m_tss.ebp; }
-    u32 stack_ptr() const { return m_tss.esp; }
+    u32 frame_ptr() const { return get_register_dump_from_stack().ebp; }
+    u32 stack_ptr() const { return get_register_dump_from_stack().esp; }
 
-    RegisterState& get_register_dump_from_stack();
+    RegisterState& get_register_dump_from_stack() const;
 
     u16 selector() const { return m_far_ptr.selector; }
-    TSS32& tss() { return m_tss; }
-    const TSS32& tss() const { return m_tss; }
+
+    uintptr_t& stored_esp() { return m_stored_esp; }
+    uintptr_t& stored_eip() { return m_stored_eip; }
+
+    uintptr_t& cr3() { return m_cr3; }
+    const uintptr_t& cr3() const { return m_cr3; }
```

Comment:
Please expose these as getters and setters like we do throughout the rest of the codebase:

```cpp
uintptr_t stored_esp() const { return m_stored_esp; }
void set_stored_esp(uintptr_t esp) { m_stored_esp = esp; }
```

Also note that `cr3` should be an `u32` -- it is not pointer-sized, still 32-bit on x86_64 :)

Reply:
I've removed `stored_esp()`/`eip()` because these only really exist for `switch_thread` and Scheduler can already access the member variables as it is a friend class.